### PR TITLE
Runtime measurement implementation, better debugging, and better error handling for unreachable hosts.

### PR DESCRIPTION
**Pull request by bergabman**
Basic runtime measurement implemented with environment variable.

Usage
```
RUST_LOG=info ./rustscan

[2020-09-14T19:42:20Z INFO  rustscan] Portscan runtime:   44159ms
[2020-09-14T19:42:20Z INFO  rustscan] Nmap scan runtime:  240ms
[2020-09-14T19:42:20Z INFO  rustscan] RustScan runtime    44477ms 
[2020-09-14T19:42:20Z INFO  rustscan] Ports scanned       65534
[2020-09-14T19:42:20Z INFO  rustscan] It took             0ms to scan an individual port
```

Error handling clarification at unreachable hosts. Possibly fixes bug calling CIDR or bigger ip range with rustscan.

README warning addition

Basic debuggin is now possible with `RUST_LOG=debug ./rustscan` to get rich information on the spot if a scan fails for "no reason".

### CODE REVIEW

**Comment by bergabman** [comment]
Debug output example
```
[2020-09-14T20:10:21Z DEBUG rustscan] Scanner finished building: Scanner { ips: [V4(192.168.22.139)], batch_size: 3000, timeout: 2s, quiet: false, port_strategy: Serial(SerialRange { start: 1, end: 65535 }) }
[2020-09-14T20:10:21Z DEBUG rustscan::scanner] Start scanning sockets. 
    Batch size 3000
    Number of ip-s 1
    Number of ports 65535
    Targets all together 65535 
[2020-09-14T20:10:21Z DEBUG rustscan::scanner] Connection was successful, shutting down stream 192.168.22.139:22
Open 51.104.40.132:22
[2020-09-14T20:10:21Z DEBUG rustscan::scanner] Connection was successful, shutting down stream 192.168.22.139:80
Open 51.104.40.132:80
[2020-09-14T20:10:33Z DEBUG rustscan::scanner] Socket connect error: No route to host (os error 113) 192.168.22.139:18810
[2020-09-14T20:10:33Z DEBUG rustscan::scanner] Socket connect error: No route to host (os error 113) 192.168.22.139:18811
[2020-09-14T20:10:33Z DEBUG rustscan::scanner] Socket connect error: No route to host (os error 113) 192.168.22.139:18812

File: README.md [hunk 1]
```diff
@@ -302,6 +302,9 @@ This program, by default, scans 5000 ports at a time (5000 per second).
 
 This may cause damage to a server, or may make it incredibly obvious you are scanning the server.
 
+Servers as well as individual services listening on ports can block your ip completely after a huge amount of simultaneous connections/tries.
```
Comment (bee-san) [inline README.md:305]: Servers as well as individual services can block your IP completely after a huge amount of simultaneous connections/tries, Possibly causing you to see false positives. False positives are open ports at the initial scanning process later found closed when you want to do further scanning, or connect to it.

File: src/main.rs [hunk 2]
```diff
@@ -38,6 +38,7 @@ extern crate log;
 /// If you're looking for the actual scanning, check out the module Scanner
 fn main() {
     env_logger::init();
+    let startprocess = Instant::now();
```
Comment (bee-san) [inline src/main.rs:41]: The Rust standard is snake_case, please rename to start_process :)

**Review by bee-san — CHANGES_REQUESTED** [review]

**Comment by bee-san** [comment]
Can we expose your functionality to be used in a test?

File: README.md (hunk 1 above)
Comment (bergabman) [reply README.md:305]: Changed.

File: src/main.rs (hunk 2 above)
Comment (bergabman) [reply src/main.rs:41]: Corrected

**Comment by bergabman** [comment]
This PR is not conform project requirements. I will restructure it and make a better PR later. It was probably way too early for it.

**Comment by bee-san** [comment]
Don't be so hard on yourself! It was just some minor details -- it's my fault for not including those in our wiki :(

**Pull request closed without merging**